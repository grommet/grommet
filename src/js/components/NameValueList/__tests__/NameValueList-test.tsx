import React from 'react';
import 'jest-styled-components';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';
import { axe } from 'jest-axe';

import { render } from '@testing-library/react';

import { ThemeType } from '../../../themes';
import { Grommet } from '../../Grommet';
import { NameValueList } from '..';
import { NameValuePair } from '../../NameValuePair';

const data = {
  name: 'entry',
  location: 'San Francisco',
  health: 80,
};

describe('NameValueList', () => {
  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <NameValueList>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  test(`should render`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render outside grommet wrapper`, () => {
    const { container } = render(
      <NameValueList>
        {Object.entries(data).map(([name, value]) => (
          <NameValuePair key={name} name={name}>
            {value}
          </NameValuePair>
        ))}
      </NameValueList>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render correct width of name`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList nameProps={{ width: 'xsmall' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render correct width of value`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList valueProps={{ width: 'xsmall' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render correct width of value when value is array`, () => {
    const { asFragment } = render(
      <Grommet>
        <NameValueList valueProps={{ width: 'xsmall' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test(`should render correct nameProps width with max and min`, () => {
    const { asFragment } = render(
      <Grommet>
        <NameValueList
          nameProps={{ width: { min: 'xsmall', max: 'max-content' } }}
        >
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test(`should render correct valueProps width with max and min`, () => {
    const { asFragment } = render(
      <Grommet>
        <NameValueList valueProps={{ width: { min: 'small', max: 'medium' } }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test(`should render correct alignment of name`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList nameProps={{ align: 'end' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render correct alignment of value`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList valueProps={{ align: 'start' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render pairs in a grid when layout="grid"`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList layout="grid">
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should accept and apply Grid props`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList
          align="center"
          justify="center"
          gap={{ row: 'large', column: 'xlarge' }}
        >
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render name/value as a column when pairProps = { direction: 
    'column' }`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList pairProps={{ direction: 'column' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render value above name when pairProps = { direction: 
    'column-reverse' }`, () => {
    const { container } = render(
      <Grommet>
        <NameValueList pairProps={{ direction: 'column-reverse' }}>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should render correct gap between rows and columns`, () => {
    const customGapTheme: ThemeType = {
      nameValueList: {
        gap: { column: 'small', row: 'large' },
      },
    };

    const { container } = render(
      <Grommet theme={customGapTheme}>
        <NameValueList>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should support custom theme`, () => {
    const customTheme: ThemeType = {
      nameValueList: {
        gap: { column: 'small', row: 'large' },
        pair: {
          column: {
            gap: {
              column: 'medium',
              row: 'small',
            },
          },
        },
        name: {
          width: 'xsmall',
        },
        value: {
          width: 'small',
        },
      },
      nameValuePair: {
        name: {
          color: 'brand',
          weight: 'bold',
        },
        value: {
          weight: 'lighter',
        },
      },
    };

    const { container } = render(
      <Grommet theme={customTheme}>
        <NameValueList>
          {Object.entries(data).map(([name, value]) => (
            <NameValuePair key={name} name={name}>
              {value}
            </NameValuePair>
          ))}
        </NameValueList>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
