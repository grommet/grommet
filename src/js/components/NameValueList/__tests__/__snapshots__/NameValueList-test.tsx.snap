// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NameValueList should accept and apply Grid props 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  align-items: center;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 48px 96px;
  justify-items: center;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #555555;
  font-weight: undefined;
}

.c3 {
  margin-bottom: 3px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #2e2e2e;
  font-weight: 500;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render correct alignment of name 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #555555;
  font-weight: undefined;
}

.c3 {
  margin-bottom: 3px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #2e2e2e;
  font-weight: 500;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render correct alignment of value 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c3 {
  margin-bottom: 3px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #2e2e2e;
  font-weight: 500;
}

.c4 {
  margin: 0px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #555555;
  font-weight: undefined;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render correct gap between rows and columns 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render correct nameProps width with max and min 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
    class="c0"
  >
    <dl
      class="c1"
    >
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          name
        </dt>
        <dd
          class="c4"
        >
          entry
        </dd>
      </div>
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          location
        </dt>
        <dd
          class="c4"
        >
          San Francisco
        </dd>
      </div>
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          health
        </dt>
        <dd
          class="c4"
        >
          80
        </dd>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`NameValueList should render correct valueProps width with max and min 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(min(192px, 100%), 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
    class="c0"
  >
    <dl
      class="c1"
    >
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          name
        </dt>
        <dd
          class="c4"
        >
          entry
        </dd>
      </div>
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          location
        </dt>
        <dd
          class="c4"
        >
          San Francisco
        </dd>
      </div>
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          health
        </dt>
        <dd
          class="c4"
        >
          80
        </dd>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`NameValueList should render correct width of name 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render correct width of value 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 96px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render correct width of value when value is array 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 96px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
    class="c0"
  >
    <dl
      class="c1"
    >
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          name
        </dt>
        <dd
          class="c4"
        >
          entry
        </dd>
      </div>
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          location
        </dt>
        <dd
          class="c4"
        >
          San Francisco
        </dd>
      </div>
      <div
        class="c2"
      >
        <dt
          class="c3"
        >
          health
        </dt>
        <dd
          class="c4"
        >
          80
        </dd>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`NameValueList should render name/value as a column when pairProps = { direction: \\n    'column' } 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c3 {
  margin-bottom: 3px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #2e2e2e;
  font-weight: 500;
}

.c4 {
  margin: 0px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #555555;
  font-weight: undefined;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render outside grommet wrapper 1`] = `
.c0 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: 192px minmax(auto, 384px);
  grid-gap: 12px 48px;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: bold;
}

.c2 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

@media only screen and (max-width: 768px) {
  .c0 {
    margin: 0px;
  }
}

<dl
  class="c0"
>
  <dt
    class="c1"
  >
    name
  </dt>
  <dd
    class="c2"
  >
    entry
  </dd>
  <dt
    class="c1"
  >
    location
  </dt>
  <dd
    class="c2"
  >
    San Francisco
  </dd>
  <dt
    class="c1"
  >
    health
  </dt>
  <dd
    class="c2"
  >
    80
  </dd>
</dl>
`;

exports[`NameValueList should render pairs in a grid when layout="grid" 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #555555;
  font-weight: undefined;
}

.c3 {
  margin-bottom: 3px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #2e2e2e;
  font-weight: 500;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should render value above name when pairProps = { direction: \\n    'column-reverse' } 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 384px) );
  grid-gap: 24px 48px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c3 {
  margin: 0px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #555555;
  font-weight: undefined;
}

.c4 {
  margin-bottom: 3px;
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #2e2e2e;
  font-weight: 500;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dd
        class="c3"
      >
        entry
      </dd>
      <dt
        class="c4"
      >
        name
      </dt>
    </div>
    <div
      class="c2"
    >
      <dd
        class="c3"
      >
        San Francisco
      </dd>
      <dt
        class="c4"
      >
        location
      </dt>
    </div>
    <div
      class="c2"
    >
      <dd
        class="c3"
      >
        80
      </dd>
      <dt
        class="c4"
      >
        health
      </dt>
    </div>
  </dl>
</div>
`;

exports[`NameValueList should support custom theme 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  margin: 0px;
  grid-template-columns: repeat( auto-fit,minmax(auto, 192px) );
  grid-gap: 12px 24px;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
  color: #444444;
  font-weight: lighter;
}

.c3 {
  margin-bottom: 3px;
  font-size: 18px;
  line-height: 24px;
  color: #7D4CDB;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 0px;
  }
}

<div
  class="c0"
>
  <dl
    class="c1"
  >
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        name
      </dt>
      <dd
        class="c4"
      >
        entry
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        location
      </dt>
      <dd
        class="c4"
      >
        San Francisco
      </dd>
    </div>
    <div
      class="c2"
    >
      <dt
        class="c3"
      >
        health
      </dt>
      <dd
        class="c4"
      >
        80
      </dd>
    </div>
  </dl>
</div>
`;
